### ISSUE TITLE
DRY up schemes

### ISSUE THREAD

**Issue by krancour** [akuity/kargo#680]
#649 called attention to how non-DRY we are WRT schemes for k8s APIs, but we decided not to use the approach in that PR in favor of something that addresses other underlying smells that promoted that non-DRYness.

More detail...

Each of Kargo's main components (API server, controller, garbage collector, etc.) interacts with various k8s APIs. The exact list of APIs is different from component to component. Sometimes in the course of crafting tests, we need to construct a scheme. We have thus far tended to _re_-enumerate the necessary APIs on a case by case basis. This feels onerous and brittle.

Our goal should be that each high level component enumerates the k8s APIs it uses only once in a package private helper or, at least, in a package that is clearly exclusive to the component in question. In places where this isn't currently tenable for whatever reason, it's a smell indicative of poor package structure. With minimal refactoring we can improve package structure to a point where we can also reasonably DRY things up.

This is a code hygiene/devx fix and feels low priority.

Let's tackle this _after_ v0.1.0.

**Comment by krancour** [comment]
Not stale. I want to come back to this when we have more time for addressing minor tech debt.
